test/suite/test_key_provider_disagg01.py

```diff
@@ -0,0 +1,142 @@
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+import re
+import wttest
+from helper_disagg import DisaggConfigMixin, disagg_test_class, gen_disagg_storages
+from helper import simulate_crash_restart
+from wtdataset import SimpleDataSet
+import sqlite3
+
+from wtscenario import make_scenarios
+
+# test_key_provider_disagg01.py
+#    Test basic key provider scenarios.
+@disagg_test_class
+class test_key_provider_disagg01(wttest.WiredTigerTestCase):
+    conn_base_config = ',create,statistics=(all),statistics_log=(wait=1,json=true,on_close=true),'
+    def conn_config(self):
+        return self.extensionsConfig() + self.conn_base_config + 'disaggregated=(role="leader")'
+
+    disagg_storages = gen_disagg_storages('test_key_provider_disagg01', disagg_only = True)
+
+    crash_value = [
+        ('reopen', dict(crash=False)),
+        ('crash', dict(crash=True)),
+    ]
+
+    key_rotate = [
+        # FIXME-WT-16055: Re-enable once startup or checkpoint pick-up.
+        #('no_key_rotate', dict(key_expire=-1)), # Never perform key rotation
+        ('key_rotate', dict(key_expire=0)) # Always perform key rotation
+    ]
+
+    disagg_storages = gen_disagg_storages('test_layered17', disagg_only = True)
+    scenarios = make_scenarios(disagg_storages, key_rotate, crash_value)
+
+    sqlite_meta_cursor = None
+    sqlite_key_provider_cursor = None
+    nentries = 1000
+    current_lsn = 0
+
+    MAIN_KEK_PAGE_ID = 1
+    EXPECTED_KEK_VERSION = 1
+
+    uri = "layered:test_key_provider_disagg01"
+
+    # Load the key provider store extension.
+    def conn_extensions(self, extlist):
+        config = f'=(early_load=true,config=\"verbose=-1,key_expires={self.key_expire}\")'
+        extlist.extension('test', "key_provider" + config)
+        DisaggConfigMixin.conn_extensions(self, extlist)
+
+    def validate_number_elements(self):
+        self.sqlite_meta_cursor.execute("SELECT COUNT(*) FROM pages")
+        (meta_count,) = self.sqlite_meta_cursor.fetchone()
+
+        self.sqlite_key_provider_cursor.execute("SELECT COUNT(*) FROM pages")
+        (key_provider_count,) = self.sqlite_key_provider_cursor.fetchone()
+        if (self.key_expire == 0):
+            self.assertEqual(key_provider_count, meta_count)
+        else:
+            self.assertGreaterEqual(key_provider_count, meta_count)
+
+    def validate_meta_file(self):
+        self.sqlite_meta_cursor.execute("SELECT * FROM pages ORDER BY lsn DESC LIMIT 1")
+        result = self.sqlite_meta_cursor.fetchone()
+        m = re.search(".*page_id=(\d+),lsn=(\d+).*version=(\d+)", result[-1].decode("utf-8"))
```

Comment:
I would search for each config k=v pair separately. Config format doesn't require special order, so the test should follow it.
Also, if you use named groups, then you won't need `m.group(<num>)` later, but could use `m` as a regular python dictionary.

Reply:
Sure I have done that now let me know what you think.